src/device.rs

```diff
@@ -91,6 +91,16 @@ impl<'a> Device<'a> {
         })
     }
 
+    /// Returns the parent device.
+    pub fn parent(&self) -> Option<Device> {
+        let ptr = unsafe { ::ffi::udev_device_get_parent(self.as_ptr())};
+        if ptr.is_null() {
+            None
+        } else {
+            Some(Device { _context: self._context, device: ptr})
```

Comment:
Adding the following code to the `list_devices` example results in a crash:

```rust
if let Some(parent) = device.parent() {
    println!("     parent: {:?}", parent.syspath());
}
else {
    println!("     parent: None");
}
```

The output on stderr is:

```
*** Error in `./target/debug/examples/list_devices': free(): invalid pointer: 0x00007f9c0e54aeb8 ***
Aborted
```

Adding `udev_devic_ref(ptr)` before returning fixes it with no memory leaks detected by valgrind.